### PR TITLE
feat(admin): add retry functionality to AdminAgentRunsController and …

### PR DESCRIPTION
**Pull request by tomer-tgp**
…service

- Introduced a new retry endpoint in AdminAgentRunsController to re-dispatch agent runs.
- Updated AdminAgentRunsService to handle the retry logic, including validation for clerk_user_id and dispatching runs.
- Enhanced tests for both the controller and service to cover the new retry functionality and its edge cases.
- Updated Admin module to include necessary imports for the new features.

---

> [!NOTE]
> **Medium Risk**
> Retry triggers new experiment runs and SQS dispatch from stored params; impact is limited by M2M-only access and narrow experiment-type validation.
> 
> **Overview**
> Adds an **M2M-only** admin API to **retry a past agent run** by creating a **new** run and queue dispatch, leaving the original row unchanged.
> 
> `POST admin/agent-runs/:runId/retry` is wired through `AdminAgentRunsController` with `AgentRunSchema` on the response. `AdminAgentRunsService.retry` loads the stored run, requires `clerk_user_id` in params and a **dispatchable** experiment type (currently only `compliance_setup`), then calls `ExperimentRunsService.dispatchRun` with the saved org, type, and params. Missing `clerk_user_id` or unsupported types return **400**; environments without dispatch configured return **502** when `dispatchRun` does not return a run. `AdminModule` now imports `AgentExperimentsModule` so the service can dispatch. Controller and service tests cover the new endpoint, guards, and retry edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c7a013d423fb5feeff67705c243473e98ada373f. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/admin/agentRuns/services/adminAgentRuns.service.ts [hunk 1]
```diff
@@ -95,6 +126,47 @@ export class AdminAgentRunsService extends createPrismaBase(
...
+      throw new BadRequestException(
+        'run params carry no clerk_user_id; cannot re-dispatch as the candidate',
+      )
+    }
+
+    if (!isDispatchableExperimentType(run.experimentType)) {
+      throw new BadRequestException(
+        `experiment type "${run.experimentType}" cannot be re-dispatched`,
+      )
+    }
```
[the remark this answers is not in the document]
Comment (tomer-tgp) [reply src/admin/agentRuns/services/adminAgentRuns.service.ts:154]: As designed

File: src/admin/agentRuns/services/adminAgentRuns.service.ts [hunk 2]
```diff
@@ -95,6 +127,65 @@ export class AdminAgentRunsService extends createPrismaBase(
...
+  async retry(runId: string): Promise<ExperimentRun> {
+    const run = await this.model.findUniqueOrThrow({ where: { runId } })
+
+    // A RUNNING run is still in flight; re-dispatching it would spawn a second
+    // parallel worker on the same params (duplicate domain buy / TCR submit).
+    if (run.status === ExperimentRunStatus.RUNNING) {
+      throw new ConflictException(
+        'run is still RUNNING; only finished runs can be retried',
+      )
+    }
```
[the remark this answers is not in the document]
Comment (tomer-tgp) [reply src/admin/agentRuns/services/adminAgentRuns.service.ts:141]: as designed won't fix

File: src/admin/agentRuns/services/adminAgentRuns.service.test.ts [hunk 3]
```diff
@@ -14,6 +16,7 @@ const makeRun = (overrides: Partial<ExperimentRun> = {}): ExperimentRun => ({
     candidate_first_name: 'Ada',
     candidate_last_name: 'Lovelace',
     clerk_user_id: 'user_abc',
+    trigger: 'initial',
   } as Prisma.JsonValue,
   artifactBucket: 'agent-artifacts',
   artifactKey: 'compliance_setup/run-1/artifact.json',
```
[the remark this answers is not in the document]
Comment (tomer-tgp) [reply src/admin/agentRuns/services/adminAgentRuns.service.test.ts:23]: as designed won't fix
